src/main/java/org/opensearch/security/action/tenancy/MultiTenancyConfigurationsUpdateTransportAction.java

```diff
@@ -0,0 +1,146 @@
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Set;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import org.opensearch.action.ActionListener;
+import org.opensearch.action.index.IndexResponse;
+import org.opensearch.action.support.ActionFilters;
+import org.opensearch.action.support.HandledTransportAction;
+import org.opensearch.client.Client;
+import org.opensearch.common.inject.Inject;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.opensearch.security.configuration.ConfigurationRepository;
+import org.opensearch.security.dlic.rest.api.AbstractApiAction;
+import org.opensearch.security.securityconf.impl.CType;
+import org.opensearch.security.securityconf.impl.SecurityDynamicConfiguration;
+import org.opensearch.security.securityconf.impl.v7.ConfigV7;
+import org.opensearch.security.support.ConfigConstants;
+import org.opensearch.tasks.Task;
+import org.opensearch.threadpool.ThreadPool;
+import org.opensearch.transport.TransportService;
+
+public class MultiTenancyConfigurationsUpdateTransportAction extends HandledTransportAction<MultiTenancyConfigurationsUpdateRequest, MultiTenancyConfigurationsRetrieveResponse> {
+
+    private static final Logger log = LogManager.getLogger(org.opensearch.security.action.tenancy.MultiTenancyConfigurationsUpdateTransportAction.class);
+
+    private final String securityIndex;
+    private final ConfigurationRepository config;
+    private final Client client;
+    private final ThreadPool pool;
+
+    @Inject
+    public MultiTenancyConfigurationsUpdateTransportAction(final Settings settings,
+                                             final TransportService transportService,
+                                             final ActionFilters actionFilters,
+                                             final ConfigurationRepository config,
+                                             final ThreadPool pool,
+                                             final Client client) {
+        super(MultiTenancyConfigurationsUpdateAction.NAME, transportService, actionFilters, MultiTenancyConfigurationsUpdateRequest::new);
+
+        this.securityIndex = settings.get(ConfigConstants.SECURITY_CONFIG_INDEX_NAME, ConfigConstants.OPENDISTRO_SECURITY_DEFAULT_CONFIG_INDEX);
+
+        this.config = config;
+        this.client = client;
+        this.pool = pool;
+    }
+
+    /** Load the configuration from the security index and return a copy */
+    protected final SecurityDynamicConfiguration<?> load() {
+        return config.getConfigurationsFromIndex(Collections.singleton(CType.CONFIG), false).get(CType.CONFIG).deepClone();
+    }
+
+    public Set<String> getAllConfiguredTenantNames() {
+
+        return this.config.getConfiguration(CType.TENANTS).getCEntries().keySet();
+    }
+
+    protected void validate(ConfigV7 updatedConfig) throws RuntimeException {
```

Comment:
RuntimeException by definition isn't a checked exception, please remove
```suggestion
    protected void validate(ConfigV7 updatedConfig) {
```